expression/builtin_miscellaneous.go

```diff
@@ -79,55 +78,52 @@ type sleepFunctionClass struct {
 
 func (c *sleepFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
 	err := errors.Trace(c.verifyArgs(args))
-	bt := &builtinSleepSig{newBaseBuiltinFunc(args, ctx)}
-	bt.deterministic = false
-	return bt.setSelf(bt), errors.Trace(err)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpInt)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	sig := &builtinSleepSig{baseIntBuiltinFunc{bf}}
+	sig.deterministic = false
+	return sig.setSelf(sig), errors.Trace(err)
```

Comment:
just return `sig.setSelf(sig), nil` is ok, since `err` is nil here and no need to trigger another function call.